### PR TITLE
XW-2905 | make sure list has always 2 elements + fix and simplify domain regexp

### PR DESCRIPTION
**Pull request by xkxd**
https://wikia-inc.atlassian.net/browse/XW-2905

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php [hunk 1]
```diff
@@ -241,7 +241,9 @@ private function getArticleURL( $sArticleTitle, $cityId ) {
 
 		if ( $sArticleTitle !== false ) {
 			$sArticleTitle = str_replace( ' ', '_', $sArticleTitle );
-			list($sArticleTitle, $sArticleAnchor) = explode('#', $sArticleTitle);
+			// some titles does not have a `#` in it, which break string splitting
```
Comment (bkoval) [inline extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php]: *breaks

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php [hunk 2]
```diff
@@ -241,7 +241,9 @@ private function getArticleURL( $sArticleTitle, $cityId ) {
...
+			// so we want to make sure there are exactly 2 items in array
+			list($sArticleTitle, $sArticleAnchor) = array_pad(explode('#', $sArticleTitle, 2), 2, '');
```
Comment (latata) [inline extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php]: spaces

**Comment by rybmat** [comment]
lgtm, just fix unittests

File: extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php [hunk 3]
```diff
@@ -114,7 +114,7 @@ public function getWikiDomain( $sCurrentUrl ) {
 
 		if ( isset( $aParsed['host'] ) ) {
 			$sHost = $aParsed['host'];
-			$regExp = "/(sandbox.{3}\.|preview\.|verify\.)?(([a-z]{2,3}|[a-z]{2}\-[a-z]{2})\.)?([^\.]+\.)([^\.]+\.)(.*)/i";
+			$regExp = "/(sandbox-.+?|preview|verify)\.?(([a-z]{2,3}|[a-z]{2}\-[a-z]{2})\.)?([^\.]+\.)([^\.]+\.)(.*)/i";
```
Comment (bkoval) [inline extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php]: What kind of fix was applied here?

If we fixed something, maybe we could extending unit tests to cover the cases that were previously broken?

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/WikiaInYourLang/tests/WikiaInYourLangControllerTest.php [hunk 4]
```diff
@@ -35,6 +35,8 @@ public function getDomainsList() {
 			['http://sandbox-s3.naruto.wikia.com', 'naruto.wikia.com'],
 			['http://sandbox-s3.zh.naruto.wikia.com', 'naruto.wikia.com'],
 			['http://sandbox-s3.pt-br.naruto.wikia.com', 'naruto.wikia.com'],
+			['http://sandbox-mercury.pt-br.naruto.wikia.com', 'naruto.wikia.com'],
```
Comment (bkoval) [inline extensions/wikia/WikiaInYourLang/tests/WikiaInYourLangControllerTest.php]: To be in sync with what was here before, let's cover all three domain cases (zh, pt-br, no language domain) for each sandbox address you've added.

File: extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php [hunk 5]
```diff
@@ -241,7 +241,9 @@ private function getArticleURL( $sArticleTitle, $cityId ) {
 
 		if ( $sArticleTitle !== false ) {
 			$sArticleTitle = str_replace( ' ', '_', $sArticleTitle );
-			list($sArticleTitle, $sArticleAnchor) = explode('#', $sArticleTitle);
+			// some titles does not have a `#` in it, which breaks string splitting
```
Comment (bkoval) [inline extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php]: DO not have ;> 🚓 👮 👨‍🎓 ❤️ 🔠

**Review by latata — APPROVED** [review]

File: extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php (hunk 5 above)
Comment (xkxd) [reply extensions/wikia/WikiaInYourLang/WikiaInYourLangController.class.php]: (╯°□°）╯︵ ┻━┻

**Review by hakubo — APPROVED** [review]

**Pull request merged**